pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -18,45 +18,78 @@ package v1
-			if ref.Name != "" {
-				errs = errs.Also(apis.ErrMultipleOneOf("name", "resolver"))
-			}
-		}
-		if ref.Params != nil {
+	case refResolver != "" || refParams != nil:
+		if refParams != nil {
 			errs = errs.Also(config.ValidateEnabledAPIFields(ctx, "resolver params", config.BetaAPIFields).ViaField("params"))
-			if ref.Name != "" {
+			if refName != "" {
 				errs = errs.Also(apis.ErrMultipleOneOf("name", "params"))
 			}
-			if ref.Resolver == "" {
+			if refResolver == "" {
 				errs = errs.Also(apis.ErrMissingField("resolver"))
 			}
-			errs = errs.Also(ValidateParameters(ctx, ref.Params))
+			errs = errs.Also(ValidateParameters(ctx, refParams))
+		}
+		if refResolver != "" {
+			errs = errs.Also(config.ValidateEnabledAPIFields(ctx, "resolver", config.BetaAPIFields).ViaField("resolver"))
+			if refName != "" {
+				// make sure that the name is url-like.
+				err := RefNameLikeUrl(refName)
+				if err == nil && !config.FromContextOrDefaults(ctx).FeatureFlags.EnableConciseResolverSyntax {
+					// If name is url-like then concise resolver syntax must be enabled
+					errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("feature flag %s should be set to true to use concise resolver syntax", config.EnableConciseResolverSyntax), ""))
+				}
+				if err != nil {
+					errs = errs.Also(apis.ErrInvalidValue(err, "name"))
+				}
+			}
 		}
-	case ref.Name != "":
-		// ref name must be a valid k8s name
-		if errSlice := validation.IsQualifiedName(ref.Name); len(errSlice) != 0 {
-			errs = errs.Also(apis.ErrInvalidValue(strings.Join(errSlice, ","), "name"))
+	case refName != "":
+		// ref name can be a Url-like format.
+		if err := RefNameLikeUrl(refName); err == nil {
+			// If name is url-like then concise resolver syntax must be enabled
+			if !config.FromContextOrDefaults(ctx).FeatureFlags.EnableConciseResolverSyntax {
+				errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("feature flag %s should be set to true to use concise resolver syntax", config.EnableConciseResolverSyntax), ""))
+			}
+			// In stage1 of concise remote resolvers syntax, this is a required field.
+			// TODO: remove this check when implementing stage 2 where this is optional.
+			if refResolver == "" {
+				errs = errs.Also(apis.ErrMissingField("resolver"))
+			}
+			// Or, it must be a valid k8s name
+		} else {
+			// ref name must be a valid k8s name
+			if errSlice := validation.IsQualifiedName(refName); len(errSlice) != 0 {
+				errs = errs.Also(apis.ErrInvalidValue(strings.Join(errSlice, ","), "name"))
+			}
 		}
 	default:
 		errs = errs.Also(apis.ErrMissingField("name"))
 	}
 	return errs
 }
+
+// Validate ensures that a supplied Ref field is populated
+// correctly. No errors are returned for a nil Ref.
+func (ref *Ref) Validate(ctx context.Context) (errs *apis.FieldError) {
+	if ref == nil {
+		return errs
+	}
+	return validateRef(ctx, ref.Name, ref.Resolver, ref.Params)
+}
+
+func RefNameLikeUrl(name string) error {
```

Comment:
nit: shall we have docString for the exported funcs?

Reply:
sure.

Reply:
Fixed